src/main/java/com/prgrms/needit/domain/center/service/CenterService.java

```diff
@@ -0,0 +1,72 @@
+package com.prgrms.needit.domain.center.service;
+
+import com.prgrms.needit.common.error.ErrorCode;
+import com.prgrms.needit.common.error.exception.NotFoundCenterException;
+import com.prgrms.needit.domain.center.dto.CenterCreateRequest;
+import com.prgrms.needit.domain.center.dto.CenterDetailResponse;
+import com.prgrms.needit.domain.center.dto.CenterUpdateRequest;
+import com.prgrms.needit.domain.center.entity.Center;
+import com.prgrms.needit.domain.center.repository.CenterRepository;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class CenterService {
+
+	private final CenterRepository centerRepository;
+	private final PasswordEncoder passwordEncoder;
+
+	public CenterService(
+		CenterRepository centerRepository,
+		PasswordEncoder passwordEncoder
+	) {
+		this.centerRepository = centerRepository;
+		this.passwordEncoder = passwordEncoder;
+	}
+
+	@Transactional
+	public Long createCenter(CenterCreateRequest centerRequest) {
+		return centerRepository
+			.save(centerRequest.toEntity(passwordEncoder.encode(centerRequest.getPassword())))
+			.getId();
+	}
+
+	@Transactional(readOnly = true)
+	public CenterDetailResponse getCenter(Long centerId) {
+		return centerRepository
+			.findById(centerId)
+			.map(CenterDetailResponse::new)
+			.orElseThrow(
+				() -> new NotFoundCenterException(ErrorCode.NOT_FOUND_CENTER));
+	}
+
+	@Transactional(readOnly = true)
+	public Center findActiveCenter(Long centerId) {
+		return centerRepository
+			.findById(centerId)
+			.orElseThrow(
+				() -> new NotFoundCenterException(ErrorCode.NOT_FOUND_CENTER));
+	}
+
+	@Transactional
+	public Long updateCenter(Long centerId, CenterUpdateRequest request) {
+		Center activeCenter = findActiveCenter(centerId);
+		activeCenter.changeCenterInfo(
+			request.getEmail(),
+			passwordEncoder.encode(request.getPassword()),
+			request.getName(),
+			request.getContact(),
+			request.getAddress(),
+			request.getProfileImageUrl(),
+			request.getOwner()
+		);
+		return activeCenter.getId();
+	}
+
+	@Transactional
+	public void deleteCenter(Long centerId) {
+		Center activeCenter = findActiveCenter(centerId);
+		activeCenter.deleteEntity();
+	}
+}
```

Comment:
```suggestion
	@Transactional
	public Long updateCenter(Long centerId, CenterUpdateRequest request) {
		Center activeCenter = findActiveCenter(centerId);
		activeCenter.changeCenterInfo(
			request.getEmail(),
			passwordEncoder.encode(request.getPassword()),
			request.getName(),
			request.getContact(),
			request.getAddress(),
			request.getProfileImageUrl(),
			request.getOwner()
		);
		return activeCenter.getId();
	}

	@Transactional
	public void deleteCenter(Long centerId) {
		Center activeCenter = findActiveCenter(centerId);
		activeCenter.deleteEntity();
	}
	
	@Transactional(readOnly = true)
	public Center findActiveCenter(Long centerId) {
		return centerRepository
			.findByIdAndIsDeletedFalse(centerId)
			.orElseThrow(
				() -> new NotFoundCenterException(ErrorCode.NOT_FOUND_CENTER));
	}
}
```
공통으로 사용하는 메서드는 맨 아래에 위치시켜주는 게 더 보기 좋을 것 같고 삭제되지 않은 센터를 찾아야하니 쿼리문 위 처럼 수정해주세용

Reply:
아 공통으로 사용하는 메서드는 아래에 위치하는 게 더 보기 좋은 거였군요! 공통인 부분은 다 위로 올렸던 거 같은데 하나 배워갑니당!
말씀 주신 부분 수정하겠슴다~